src/utilities.f90

```diff
@@ -812,6 +813,50 @@ subroutine deriv_2d(x,y,z,zxp,zyp)
 
 end subroutine deriv_2d
 
+subroutine deriv_3d(r,phi,z,f,frp,fphip,fzp)
```

Comment:
Could you change the order of the arguments to be have phi be last to be consistent with the arrays.